acceptance-tests/src/main/java/io/blueocean/ath/SmartWebElement.java

```diff
@@ -66,78 +46,25 @@ protected WebDriver getDriver() {
     }
 
     /**
-     * Gets elements
-     * @return the elements found
+     * @return a FluentWait with default polling / timeout / exception handling
      */
-    public List<WebElement> getElements() {
-        return new FluentWait<>(getDriver())
-            .pollingEvery(100, TimeUnit.MILLISECONDS)
-            .withTimeout(DEFAULT_TIMEOUT, TimeUnit.MILLISECONDS)
-            .ignoring(NoSuchElementException.class)
-            .until(ExpectedConditions.numberOfElementsToBeMoreThan(by, 0));
-    }
-
-    /**
-     * Gets the first matching element
-     * @return see description
-     */
-    public WebElement getElement() throws NoSuchElementException {
-        List<WebElement> elements = getElements();
-        if (elements == null || elements.isEmpty()) {
-            throw new NoSuchElementException("Nothing matched for: " + expr);
-        }
-        if (elements.size() > 1) {
-            throw new NoSuchElementException("Too many elements returned for: " + expr);
-        }
-        return elements.get(0);
-    }
-
-    /**
-     * Iterates over all found elements with the given function
-     * @param fn to perform on the elements
-     */
-    public void forEach(java.util.function.Consumer<WebElement> fn) {
-        for (WebElement e : getElements()) {
-            fn.accept(e);
-        }
-    }
-
-    /**
-     * Determines if the element is visible
-     * @return true if visible, false if not
-     */
-    public boolean isVisible() {
-        return getElement().isDisplayed();
-    }
-
-    /**
-     * Determines if the element is present
-     * @return true if present, false if not
-     */
-    public boolean isPresent() {
-        try {
-            if (getDriver().findElement(by) == null) {
-                return false;
-            }
-            return true;
-        } catch(NoSuchElementException e) {
-            return false;
-        }
+    private FluentWait<WebDriver> buildWait() {
+        return LocalDriver.buildWait();
     }
 
     @Override
     public void click() {
         for (int i = 0; i < RETRY_COUNT; i++) {
             try {
-                WebElement e = getElement();
+                WebElement e = buildWait().until(ExpectedConditions.elementToBeClickable(element));
```

Comment:
This got lost in #1484 and needs to be there to make sure we don't click on elements that are transitioning from disabled to enabled. Frequent cause of flakiness in creation tests when robot picks a repo and clicks "Create Pipeline" while it's still disabled, resulting in a hung test.

Reply:
interesting

Reply:
right, definitely need this change after talking with Cliff, this was a stupid mistake on my part